packages/@aws-cdk/aws-cognito-identitypool/lib/identitypool-role-attachment.ts

```diff
@@ -176,6 +183,17 @@ export class IdentityPoolRoleAttachment extends Resource implements IIdentityPoo
   ): { [name:string]: CfnIdentityPoolRoleAttachment.RoleMappingProperty } | undefined {
     if (!props || !props.length) return undefined;
     return props.reduce((acc, prop) => {
+      let mappingKey;
+      if (prop.mappingKey) {
+        mappingKey = prop.mappingKey;
+      } else {
+        const providerUrl = prop.providerUrl.value;
+        if (Token.isUnresolved(providerUrl)) {
+          throw new Error('mappingKey must be provided when providerUrl.value is an unresolved token');
```

Comment:
I'm happy to make this change; that was wording I've copied from other parts of the CDK such as https://github.com/aws/aws-cdk/blob/74318c7d22bfc00de9e005f68a0a6aaa58c7db39/packages/%40aws-cdk/aws-ssm/lib/util.ts#L51 and https://github.com/aws/aws-cdk/blob/cea1039e3664fdfa89c6f00cdaeb1a0185a12678/packages/%40aws-cdk/aws-iam/lib/principals.ts#L335.

I couldn't decide if this was a fix or a feature; I fell on the side of this being a feature as it does actually provide new functionality, rather than altering existing functionality; it's just that the new feature addresses a gap that was previously present in the CDK's capabilities. Once again, I'm happy to defer to your judgment here.

Reply:
Ah, I've made the change manually, rather than applying your commit suggestion; that's Github functionality I hadn't seen before.